File: Workspace/springBoot/src/main/java/formation/sopra/springBoot/entities/Adresse.java
package formation.sopra.springBoot.entities;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.validation.constraints.NotEmpty;

import com.fasterxml.jackson.annotation.JsonView;
import com.sun.istack.NotNull;

import formation.sopra.springBoot.entities.views.Views;

@Embeddable
public class Adresse {
	@JsonView(Views.Common.class)
	@NotNull
	private Integer numero;
	@JsonView(Views.Common.class)
	@Column(name = "rue", length = 200)
	@NotEmpty
	private String rue;
	@JsonView(Views.Common.class)
	@Column(name = "code_postal", length = 20)
	@NotEmpty
	private String codePostal;
	@JsonView(Views.Common.class)
	@Column(name = "ville", length = 150)
	@NotEmpty
	private String ville;

	public Adresse() {

	}

	public Adresse(Integer numero, String rue, String codePostal, String ville) {
		this.numero = numero;
		this.rue = rue;
		this.codePostal = codePostal;
		this.ville = ville;
	}

	public Integer getNumero() {
		return numero;
	}

	public void setNumero(Integer numero) {
		this.numero = numero;
	}

	public String getRue() {
		return rue;
	}

	public void setRue(String rue) {
		this.rue = rue;
	}

	public String getCodePostal() {
		return codePostal;
	}

	public void setCodePostal(String codePostal) {
		this.codePostal = codePostal;
	}

	public String getVille() {
		return ville;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}

}

File: Workspace/angular-final/src/app/model/adresse.ts
export class Adresse {
  constructor(
    private _numero: number = 0,
    private _rue: string = '',
    private _codePostal: string = '',
    private _ville: string = ''
  ) {}

  /**
   * Getter numero
   * @return {number }
   */
  public get numero(): number {
    return this._numero;
  }

  /**
   * Getter rue
   * @return {string }
   */
  public get rue(): string {
    return this._rue;
  }

  /**
   * Getter codePostal
   * @return {string }
   */
  public get codePostal(): string {
    return this._codePostal;
  }

  /**
   * Getter ville
   * @return {string }
   */
  public get ville(): string {
    return this._ville;
  }

  /**
   * Setter numero
   * @param {number } value
   */
  public set numero(value: number) {
    this._numero = value;
  }

  /**
   * Setter rue
   * @param {string } value
   */
  public set rue(value: string) {
    this._rue = value;
  }

  /**
   * Setter codePostal
   * @param {string } value
   */
  public set codePostal(value: string) {
    this._codePostal = value;
  }

  /**
   * Setter ville
   * @param {string } value
   */
  public set ville(value: string) {
    this._ville = value;
  }
}
